lib/experimental/class-wp-html-tokenizer-2.php

```diff
@@ -0,0 +1,492 @@
+<?php
+
+class WP_Attribute_Match {
+	private $name;
+	private $value;
+	private $start_index;
+	private $end_index;
+
+	/**
+	 * @param $name
+	 * @param $value
+	 * @param $start_index
+	 * @param $end_index
+	 */
+	public function __construct( $name, $value, $start_index, $end_index ) {
+		$this->name        = $name;
+		$this->value       = $value;
+		$this->start_index = $start_index;
+		$this->end_index   = $end_index;
+	}
+
+	public function __toString() {
+		return "{$this->name}=\"{$this->value}\"";
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public function getName() {
+		return $this->name;
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public function getValue() {
+		return $this->value;
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public function getStartIndex() {
+		return $this->start_index;
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public function getEndIndex() {
```

Comment:
another nitpick: as core PHP we will want to use names in snake_case like `get_start_index()` and `get_end_index()`

Reply:
It's reported by CI job:
https://github.com/WordPress/gutenberg/runs/7378928305?check_suite_focus=true

You can also run it locally with:

https://github.com/WordPress/gutenberg/blob/110c23d4773c50781aa330457392419d6fc2c6b6/package.json#L273

Usage is a bit unfriendly because you need to have Docker and run `npx wp-env start` first.